go/vt/dbconnpool/connection.go

```diff
@@ -36,24 +37,21 @@ func (dbc *DBConnection) handleError(err error) {
 
 // ExecuteFetch is part of PoolConnection interface.
 func (dbc *DBConnection) ExecuteFetch(query string, maxrows int, wantfields bool) (*proto.QueryResult, error) {
-	start := time.Now()
+	defer dbc.mysqlStats.Record("Exec", time.Now())
```

Comment:
Great then. i guess I was confused with:
defer f() {
  XXX
}
where XXX is evaluated when the function runs at the end, not at declaration time.